lib/application.js

```diff
@@ -604,33 +604,37 @@ Box.Application = (function() {
 		 */
 		getModuleConfig: function(element, name) {
 
-			var instanceData = getInstanceDataByElement(element),
-				configElement;
-
-			if (instanceData) {
+			var instanceData = getInstanceDataByElement(element);
+			var moduleConfig = null;
 
-				if (!instanceData.config) {
-					// <script type="text/x-config"> is used to store JSON data
-					configElement = Box.DOM.query(element, 'script[type="text/x-config"]');
+			if (instanceData && instanceData.config) {
+				// Check if we've already read the configurations before
+				moduleConfig = instanceData.config;
+			} else {
+				// Read the special script element that stores module configuration in the markup
+				var configElement = Box.DOM.query(element, 'script[type="text/x-config"]');
 
-					// <script> tag supports .text property
-					if (configElement) {
-						instanceData.config = JSON.parse(configElement.text);
-					}
+				// <script> tag supports .text property
+				if (configElement) {
+					moduleConfig = JSON.parse(configElement.text);
 				}
 
-				if (!instanceData.config) {
-					return null;
-				} else if (typeof name === 'undefined') {
-					return instanceData.config;
-				} else if (name in instanceData.config) {
-					return instanceData.config[name];
-				} else {
-					return null;
+				// Cache the configurations for performance, if the module instance has been created
+				if (instanceData) {
+					instanceData.config = moduleConfig;
```

Comment:
Would require a bit of refactor since instanceData is managed by start/stop. Probably not worth it at this point